app/screens/thread/thread.tsx

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo} from 'react';
+import {BackHandler, StyleSheet, View} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+import {Edge, SafeAreaView} from 'react-native-safe-area-context';
+
+import CompassIcon from '@components/compass_icon';
+import PostDraft from '@components/post_draft';
+import {THREAD_ACCESSORIES_CONTAINER_NATIVE_ID} from '@constants/post_draft';
+import {useTheme} from '@context/theme';
+import {useAppState, useIsTablet} from '@hooks/device';
+import {dismissModal, mergeNavigationOptions} from '@screens/navigation';
+
+import ThreadPostList from './thread_post_list';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type PostModel from '@typings/database/models/servers/post';
+
+type ThreadProps = {
+    channel: ChannelModel;
+    componentId: string;
+    rootPost: PostModel;
+};
+
+const edges: Edge[] = ['left', 'right'];
+
+const getStyleSheet = StyleSheet.create(() => ({
+    flex: {
+        flex: 1,
+    },
+}));
+
+const CLOSE_BUTTON_ID = 'close-threads';
+
+const Thread = ({channel, componentId, rootPost}: ThreadProps) => {
+    const appState = useAppState();
+    const isTablet = useIsTablet();
+    const styles = getStyleSheet();
+    const theme = useTheme();
+
+    const close = useCallback(() => {
+        dismissModal({componentId});
+        return true;
+    }, []);
+
+    const leftButton = useMemo(() => {
```

Comment:
do you really need this here?

Reply:
As we would anyway have follow button, I left this button here as well. It's not needed as it doesn't contain any JSX button. Moved to the `switchToThread`. Thank you 🙂